### PR TITLE
Add support for Robô Aspirador de Pó KaBuM! Smart 500

### PR DESCRIPTION
**Pull request by Oglaf**
Added support for a Brazilian vacuum cleaner model. I have been using it for the last two weeks without any problems.

This file uses almost all DPs suggested by local Tuya for "generic" vacuum cleaners, the only difference are IDs 25 (activate) and 29 (locate) which can be replaced by IDs 2 and 31 respectively.
It could be possible to create two files instead and use them as an attempt for "generic Tuya vacuum cleaner".

### CODE REVIEW

**Comment by make-all** [comment]
I don't know what is a "generic vacuum cleaner".  Is it based on any particular documentation or set of models that confirm, or is it just the first vacuum cleaner that localtuya implemented, and others all vary from the config in ways like you have found?

**Comment by Oglaf** [comment]
Hi Json, it is an educated guess based on a set of models that uses Tuya WR3 wifi module.

I know it is not 100% "error proof", as Tuya doesn't have a development guidance document and any customer can change it as their see fit. But there's a chance that "off-brand" wifi models manufactured between 2020-2022 can work without requesting to develop a file for specific device.

It's just a suggestion, of course that's 100% your decision.

File: custom_components/tuya_local/devices/kabum_smart_500 [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    - id: 27
+      type: string
+      name: command
+      mapping:
+        - dps_val: standby
+          value: standby
+        - dps_val: smart
+          value: Auto cleaning
+        - dps_val: spiral
+          value: Stop_cleanning
```
[the remark this answers is not in the document]
Comment (Oglaf) [reply custom_components/tuya_local/devices/kabum_smart_500:59]: Yes, that's a typo, will fix it. Strange that I used spot cleaning today without issues, "value" is used for front-end purposes?

[the remark this answers is not in the document]
Comment (make-all) [reply custom_components/tuya_local/devices/kabum_smart_500:59]: dps_val is what is in the tuya protocol, value is what is passed to HA.  In most cases, HA will pass that straight through to the front end, but I think this one goes through the translations first, so specific values will get translated nicely and picked up by some custom cards as specific commands with appropriate icons.

[the remark this answers is not in the document]
Comment (make-all) [reply custom_components/tuya_local/devices/kabum_smart_500:59]: The specific values are here: https://github.com/home-assistant/core/blob/dev/homeassistant/components/vacuum/__init__.py#L56

File: custom_components/tuya_local/devices/kabum_smart_500 [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+          value: High
+secondary_entities:
+  - entity: switch
+    name: Mute voice
+    icon: "mdi:account-voice"
+    dps:
+      - id: 31
+        type: boolean
+        name: switch
+  - entity: select
```
[the remark this answers is not in the document]
Comment (Oglaf) [reply custom_components/tuya_local/devices/kabum_smart_500:102]: Oh, I never thought about using vacuum.send_command and felt like something was off. I was using "Input select: Select" service for my automations, thanks!

File: custom_components/tuya_local/devices/kabum_smart_500 (hunk 1 above)
[the remark this answers is not in the document]
Comment (Oglaf) [reply custom_components/tuya_local/devices/kabum_smart_500:59]: Thank you for clarifying!

**Pull request merged**